### PR TITLE
Add io_type axis with default `PINNED_BUFFER` to nvbench PQ multithreaded reader

### LINKED ISSUE #16758: [FEA] Add `io_type` axis with `PINNED_BUFFER` default value to `PARQUET_MULTITHREAD_READER_NVBENCH`
**Is your feature request related to a problem? Please describe.**
Currently, the `PARQUET_MULTITHREAD_READER_NVBENCH` uses a file data source and this option cannot be changed at runtime. However, Spark-RAPIDS always uses a pinned host buffer data source with the libcudf parquet reader. We should add an `io_type` axis and give it a default setting of `PINNED_BUFFER`. This way the benchmark will more closely model the interface that Spark-RAPIDS uses with libcudf.

### PR DESCRIPTION
**Pull request by mhaseeb123**
## Description
Closes #16758

This PR adds an `io_type` axis to the benchmarks in `PARQUET_MULTITHREAD_READER_NVBENCH` with `PINNED_BUFFER` as default value. More description at #16758.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/io/parquet/parquet_reader_multithread.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ std::string get_label(std::string const& test_name, nvbench::state const& state)
 }
 
 std::tuple<std::vector<cuio_source_sink_pair>, size_t, size_t> write_file_data(
-  nvbench::state& state, std::vector<cudf::type_id> const& d_types)
+  nvbench::state& state, std::vector<cudf::type_id> const& d_types, io_type const io_source_type)
```
Comment (PointKernel) [inline cpp/benchmarks/io/parquet/parquet_reader_multithread.cpp:53]:
```suggestion
  nvbench::state& state, std::vector<cudf::type_id> const& d_types, io_type io_source_type)
```
There are [many discussions](https://stackoverflow.com/questions/117293/does-using-const-on-function-parameters-have-any-effect-why-does-it-not-affect) about it but passing const value is not encouraged in general. Not a hard requirement though.

**Review by PointKernel — APPROVED** [review]
Looks good

One non-blocking nit

File: cpp/benchmarks/io/parquet/parquet_reader_multithread.cpp (hunk 1 above)
Comment (mhaseeb123) [reply cpp/benchmarks/io/parquet/parquet_reader_multithread.cpp:53]: Makes sense. Let me remove it.

**Review by davidwendt — APPROVED** [review]

**Review by kingcrimsontianyu — APPROVED** [review]
Lgtm!

**Pull request merged**